lib/screens/routes/route.ex

```diff
@@ -157,4 +157,11 @@ defmodule Screens.Routes.Route do
   def get_color_for_route(_, :rail), do: :purple
   def get_color_for_route(_, :ferry), do: :teal
   def get_color_for_route(_, _), do: :yellow
+
+  def get_icon_from_route(%{type: :rail}), do: :cr
+  def get_icon_from_route(%{short_name: "SL" <> _}), do: :silver
+  def get_icon_from_route(%{type: :bus}), do: :bus
+  def get_icon_from_route(%{type: :ferry}), do: :ferry
+  def get_icon_from_route(%{id: id}), do: get_color_for_route(id)
+  def get_icon_from_route(_), do: :yellow
```

Comment:
Is this right? Same for the other cases that return a color—is this func supposed to return the vehicle icon for the route?

Reply:
The `FreeText`component is able to determine if a given string should show a `TextPill` or an `IconPill`. I do agree that the function name is a little misleading. I think I'll change it to `get_icon_or_color_from_route`.